### PR TITLE
Feature/cli update config and login

### PR DESCRIPTION
**Pull request by the38th**

### CODE REVIEW

File: lib/uffizzi/config_file.rb [hunk 1]
```diff
@@ -71,20 +71,43 @@ def list
...
+
       def read
-        JSON.parse(File.read(CONFIG_PATH), symbolize_names: true)
+        data = File.read(CONFIG_PATH)
+        options = data.split("\n")
+        result = {}
+        options.each do |option|
+          key, value = option.split('=', 2)
+          result[key.strip.to_sym] = value.strip
+        end
```
Comment (moklidia) [inline lib/uffizzi/config_file.rb:88]: Let's use functional methods, like `reduce`

File: lib/uffizzi/config_file.rb [hunk 2]
```diff
@@ -71,20 +71,43 @@ def list
...
+        prepared_data = prepare_data(data)
+        file.write(prepared_data)
         file.close
       end
 
+      def prepare_data(data)
+        result = ''
+        data.each_key do |key|
+          result = "#{result}#{key} = #{data[key]}\n"
+        end
```
Comment (moklidia) [inline lib/uffizzi/config_file.rb:106]: Let's use `reduce`

File: Dockerfile [hunk 3]
```diff
@@ -23,7 +23,7 @@ RUN gem install bundler -v 2.3.9
 
 COPY lib/uffizzi/version.rb ./lib/uffizzi/
 COPY uffizzi.gemspec .
-COPY Gemfile* .
+COPY Gemfile* ./
```
Comment (7R41N33) [inline Dockerfile:26]: It's unnecessary

File: lib/uffizzi/auth_helper.rb [hunk 4]
```diff
@@ -5,14 +5,13 @@ module AuthHelper
...
-          ConfigFile.option_exists?(:hostname)
+          ConfigFile.option_has_value?(:account_id) &&
+          ConfigFile.option_has_value?(:cookie) &&
+          ConfigFile.option_has_value?(:server)
       end
 
-      def project_set?
-        ConfigFile.exists? &&
-          ConfigFile.option_exists?(:project)
+      def project_set?(options)
```
Comment (7R41N33) [inline lib/uffizzi/auth_helper.rb:13]: I think it should not be here. How does this method relate to auth helpers?

Comment (7R41N33) [inline lib/uffizzi/auth_helper.rb:13]: I think it should not be here. How does this method relate to auth helpers?

File: lib/uffizzi/cli/config.rb [hunk 5]
```diff
@@ -7,13 +7,19 @@ module Uffizzi
   class CLI::Config < Thor
     include ApiClient
 
+    class << self
+      def help(_shell, _subcommand)
+        Cli::Common.show_manual(:config)
```
Comment (7R41N33) [inline lib/uffizzi/cli/config.rb:12]: Do we really need this?

Comment (7R41N33) [inline lib/uffizzi/cli/config.rb:12]: Do we really need this?

File: lib/uffizzi/cli.rb [hunk 6]
```diff
@@ -4,19 +4,24 @@
 
 module Uffizzi
   class CLI < Thor
+    require_relative 'cli/common'
+
+    class_option :help, type: :boolean, aliases: HELP_MAPPINGS
```
Comment (7R41N33) [inline lib/uffizzi/cli.rb:9]: Do we really need this?

Comment (7R41N33) [inline lib/uffizzi/cli.rb:9]: Do we really need this?

File: lib/uffizzi/cli/project/compose.rb [hunk 7]
```diff
@@ -12,25 +12,27 @@ class CLI::Project::Compose < Thor
...
     private
 
-    def run(options, command)
+    def run(command)
       return Uffizzi.ui.say('You are not logged in.') unless Uffizzi::AuthHelper.signed_in?
-      return Uffizzi.ui.say('This command needs project to be set in config file') unless Uffizzi::AuthHelper.project_set?
+      return Uffizzi.ui.say('This command needs project to be set in config file') unless Uffizzi::AuthHelper.project_set?(options)
 
+      @project_slug = options[:project].nil? ? ConfigFile.read_option(:project) : options[:project]
+      @server = ConfigFile.read_option(:server)
```
Comment (7R41N33) [inline lib/uffizzi/cli/project/compose.rb:35]: Let's avoid using such kind of variables started with '@' in this way

Comment (7R41N33) [inline lib/uffizzi/cli/project/compose.rb:35]: Let's avoid using such kind of variables started with '@' in this way

File: Dockerfile (hunk 3 above)
Comment (axisofentropy) [reply Dockerfile:26]: This is necessary when I build on my workstation. Without the foreslash, the build ends with this error:
```
Step 10/21 : COPY Gemfile* .
When using COPY with more than one source file, the destination must be a directory and end with a /
```

File: config/uffizzi.rb [hunk 8]
```diff
@@ -12,13 +12,14 @@ def self.configure
...
-    config.hostname = 'http://web:7000'
+    config.server = 'http://web:7000'
     config.credential_types = {
       dockerhub: 'UffizziCore::Credential::DockerHub',
       azure: 'UffizziCore::Credential::Azure',
       google: 'UffizziCore::Credential::Google',
       amazon: 'UffizziCore::Credential::Amazon',
       github_container_registry: 'UffizziCore::Credential::GithubContainerRegistry',
     }
+    config.default_server = 'app.uffizzi.com'
```
Comment (axisofentropy) [inline config/uffizzi.rb:23]: Think this value should be `https://app.uffizzi.com/`

**Review by axisofentropy — APPROVED** [review]
Let's get this merged! Needed for GitHub Action.

**Pull request merged**